packages/block-library/src/cover/transforms.js

```diff
@@ -64,36 +64,54 @@ const transforms = {
 		{
 			type: 'block',
 			blocks: [ 'core/group' ],
-			isMatch: ( { backgroundColor, gradient, style } ) => {
-				/*
-				 * Make this transformation available only if the Group has background
-				 * or gradient set, because otherwise `Cover` block displays a Placeholder.
-				 *
-				 * This helps avoid arbitrary decisions about the Cover block's background
-				 * and user confusion about the existence of previous content.
-				 */
-				return (
+			transform: ( attributes, innerBlocks ) => {
+				const {
+					align,
+					anchor,
+					backgroundColor,
+					gradient,
+					layout,
+					style,
+				} = attributes;
+				// If no background or gradient color is provided, default to 50% opacity.
+				// This matches the styling of a Cover block with a background image,
+				// in the state where a background image has been removed.
+				const dimRatio =
 					backgroundColor ||
 					style?.color?.background ||
-					style?.color?.gradient ||
-					gradient
-				);
-			},
-			transform: (
-				{ align, anchor, backgroundColor, gradient, style },
-				innerBlocks
-			) => {
+					style?.color?.gradient
+						? undefined
+						: 50;
+
+				const newAttributes = {
+					align,
+					anchor,
+					dimRatio,
+					overlayColor: backgroundColor,
+					customOverlayColor: style?.color?.background,
+					gradient,
+					customGradient: style?.color?.gradient,
+				};
+
+				// For variations that use a flex layout (e.g. Row and Stack),
+				// wrap the block in a Cover instead of converting directly to the cover block.
```

Comment:
I think some rewording like `preserve the Group block` or something might be a bit more clear.

Also I'm wondering if we should be consistent here and always add the `Group` block nested in the new `Cover` block. That way we can keep the other layout settings.

Finally I think it makes sense to remove existing background related attributes, if they exist, from the Group, as they are also added to the `Cover`. For example if we have a Group with background `red`, it will result in both Group and Cover to have the red background. Makes sense?